src/client/components/pivot-application/pivot-application.tsx

```diff
@@ -145,7 +145,7 @@ export class PivotApplication extends React.Component<PivotApplicationProps, Piv
     if (!dataSources.size) throw new Error('must have data sources');
     var dataSource = dataSources.first();
 
-    var essence = this.getEssenceFromHash() || Essence.fromDataSource(dataSource, dataSources, visualizations);
+    var essence = this.getEssenceFromHash() || Essence.fromDataSource(dataSource, {dataSources: dataSources, visualizations: visualizations});
```

Comment:
FYI in TypeScript (like in ES6) you can shorten:
`{dataSources: dataSources, visualizations: visualizations}`

to
`{dataSources, visualizations}`